streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -1522,6 +1525,92 @@ private void maybeScheduleFollowupRebalance(final long encodedNextScheduledRebal
         }
     }
 
+    private AssignmentError validateTaskAssignment(final ApplicationState applicationState,
+                                                   final TaskAssignment taskAssignment) {
+        final Collection<KafkaStreamsAssignment> assignments = taskAssignment.assignment();
+
+        // Check for AssignmentError.ACTIVE_TASK_ASSIGNED_MULTIPLE_TIMES
+        final Set<TaskId> activeTasks = new HashSet<>();
+        for (final KafkaStreamsAssignment assignment : assignments) {
```

Comment:
(if you're wondering how the `ACTIVE_AND_STANDBY_TASK_ASSIGNED_TO_SAME_KAFKASTREAMS` check works in that loop, it's because as long as there is a version of that taskId already assigned to that client, then we know it must be an active and standby task assigned at the same time. Since an AssignedTask with the same taskId and type is considered equivalent, and the assignment uses a Set, so there can only be one copy of a given taskId-taskType combination in each KafkaStreamsAssignment#assignment)